File: backend/app/schemas/salary.py
"""
工资记录 - 请求/响应模型；金额 API 为明文（元），数据库存储为加密

安全流程：
1. 用户输入明文金额
2. API 通过 SalaryRecordCreate 接收明文
3. Service 层使用 encrypt_amount() 加密后存储为 amount_encrypted
4. 读取时使用 decrypt_amount() 解密后通过 SalaryRecordResponse 返回明文
"""
from datetime import date, datetime
from typing import List, Optional

from pydantic import BaseModel, Field

from app.models.salary import SalaryRecord


class SalaryRecordBase(BaseModel):
    config_id: str
    amount: float = Field(..., gt=0, description="实发金额（元）- 明文接收，将在 service 层加密")
    payday_date: date
    salary_type: str = Field(default="normal", pattern="^(normal|bonus|allowance|other)$")
    images: Optional[List[str]] = None
    note: Optional[str] = None
    mood: str = Field(..., pattern="^(happy|relief|sad|angry|expect)$")


class SalaryRecordCreate(SalaryRecordBase):
    """
    创建工资记录请求
    注意：amount 字段为明文，由 salary_service.create() 使用 encrypt_amount() 加密后存储
    """


class SalaryRecordUpdate(BaseModel):
    amount: Optional[float] = Field(None, gt=0)
    payday_date: Optional[date] = None
    salary_type: Optional[str] = Field(None, pattern="^(normal|bonus|allowance|other)$")
    images: Optional[List[str]] = None
    note: Optional[str] = None
    mood: Optional[str] = Field(None, pattern="^(happy|relief|sad|angry|expect)$")


class SalaryRecordResponse(BaseModel):
    """
    工资记录响应
    注意：amount 字段已由 decrypt_amount() 解密，可安全返回
    """
    id: str
    user_id: str
    config_id: str
    amount: float  # 已解密的明文金额
    payday_date: date
    salary_type: str
    images: Optional[List[str]] = None
    note: Optional[str] = None
    mood: str
    risk_status: str
    created_at: datetime
    updated_at: datetime

File: miniapp/src/api/salary.ts
/**
 * 工资记录 - 与 backend /api/v1/salary 一致；金额 API 为元
 */
import request from '@/utils/request'

const PREFIX = '/api/v1/salary'

/** 工资类型 */
export type SalaryType = 'normal' | 'bonus' | 'allowance' | 'other'
/** 心情 */
export type MoodType = 'happy' | 'relief' | 'sad' | 'angry' | 'expect'

export interface SalaryRecord {
  id: string
  user_id: string
  config_id: string
  amount: number
  before_tax: number | null
  tax_amount: number | null
  payday_date: string
  salary_type: SalaryType
  salary_source: string | null
  delay_days: number | null
  delay_reason: string | null
  images: string[] | null
  note: string | null
  mood: MoodType
  risk_status: string
  created_at: string
  updated_at: string
}

export interface SalaryRecordCreate {
  config_id: string
  amount: number
  before_tax?: number | null
  tax_amount?: number | null
  payday_date: string
  salary_type?: SalaryType
  salary_source?: string | null
  delay_days?: number | null
  delay_reason?: string | null
  images?: string[] | null
  note?: string | null
  mood: MoodType
}

export interface SalaryRecordUpdate {
  amount?: number
  before_tax?: number | null
  tax_amount?: number | null
  payday_date?: string
  salary_type?: SalaryType
  salary_source?: string | null
  delay_days?: number | null
  delay_reason?: string | null
  images?: string[] | null
  note?: string | null
  mood?: MoodType
}

export interface SalaryListParams {
  config_id?: string
  from_date?: string
  to_date?: string
  limit?: number
  offset?: number
}

/** 列表 */
export function listSalary(params?: SalaryListParams) {
  let url = PREFIX
  if (params && Object.keys(params).length) {
    const queryParts: string[] = []
    if (params.config_id) queryParts.push(`config_id=${encodeURIComponent(params.config_id)}`)
    if (params.from_date) queryParts.push(`from_date=${encodeURIComponent(params.from_date)}`)
    if (params.to_date) queryParts.push(`to_date=${encodeURIComponent(params.to_date)}`)
    if (params.limit != null) queryParts.push(`limit=${encodeURIComponent(String(params.limit))}`)
    if (params.offset != null)
      queryParts.push(`offset=${encodeURIComponent(String(params.offset))}`)
    const qs = queryParts.join('&')
    if (qs) url += (url.includes('?') ? '&' : '?') + qs
  }
  return request<SalaryRecord[]>({ url, method: 'GET' })
}

/** 新增 */
export function createSalary(data: SalaryRecordCreate) {
  return request<SalaryRecord>({ url: PREFIX, method: 'POST', data })
}

/** 单条 */
export function getSalary(recordId: string) {
  return request<SalaryRecord>({ url: `${PREFIX}/${recordId}`, method: 'GET' })
}

/** 更新 */
export function updateSalary(recordId: string, data: SalaryRecordUpdate) {
  return request<SalaryRecord>({ url: `${PREFIX}/${recordId}`, method: 'PUT', data })
}

/** 删除 */
export function deleteSalary(recordId: string) {
  return request<void>({ url: `${PREFIX}/${recordId}`, method: 'DELETE' })
}
